ssl/quic/quic_txp.c

```diff
@@ -837,11 +837,7 @@ int ossl_quic_tx_packetiser_generate(OSSL_QUIC_TX_PACKETISER *txp,
          ++enc_level)
         txp_pkt_cleanup(&pkt[enc_level], txp);
 
-    /*
-     * If we already successfully did at least one, make sure we report this via
-     * the return code.
-     */
-    return pkts_done > 0 ? TX_PACKETISER_RES_SENT_PKT : res;
```

Comment:
This change is not obviously correct to me. Previously we returned `TX_PACKETISER_RES_SENT_PKT` as long as one packet was successfully committed. Now we return TX_PACKETISER_RES_FAILURE if any of them fail to commit.

Reply:
I was contemplating on whether to make this change or not but IMO failing to commit a packet indicates a serious internal error and we should not continue. This allows indicating the internal error right away. @hlandau what's your opinion?

Reply:
The original code was intentional. If we encounter an error of some kind after generating one or more packets successfully, reporting that we sent packets should take priority over reporting the error. A subsequent call can try again, fail for the same reason and report that error. So this change should be backed out.

Reply:
Why don't we change generate to use a separate `pkts_sent` output argument, which is always written on all escape paths from generate()?

Then the code under `TX_PACKETISER_RES_SENT_PKT` can be keyed on `if (pkts_sent > 0)` instead and the code under `default:` can be executed if the packetiser fails. Note that if it fails and has sent packets both of these blocks will execute, in that order.

Reply:
OK, I will do that